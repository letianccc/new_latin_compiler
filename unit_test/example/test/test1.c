


void main() {
    int a1 = 3, a2 = 2, a3 = 1;
    int b1 = 1, b2 = 3, b3;
    int c1, c2, c3;
    printf("target 1: %d\n", a3);
    b3 = 2;
    printf("target 2: %d\n", b3);

    double d1, d2, d3;
    d1 = 3.5;
    printf("target 3.5 : %f\n", d1);
    printf("target 4.5 : %f\n", 4.5);
    printf("target 5   : %d\n", 5);
    printf("target None: None\n");

    func1(1, 2);
    func2(1, 2);
    func3();
    func4(1);
    getchar();
}



void func1(int a, int b) {
    int a1 = 3, a2 = 2, a3 = 1;
    int b1 = 1, b2 = 3, b3;
    int c1, c2, c3;
    printf("target 1: %d\n", a3);
    b3 = 2;
    printf("target 2: %d\n", b3);

    double d1, d2, d3;
    d1 = 3.5;
    printf("target 3.5 : %f\n", d1);
    printf("target 4.5 : %f\n", 4.5);
    printf("target 5   : %d\n", 5);
    printf("target None: None\n");

}

void func2(int a, int b) {
    int c = a;
    printf("target 1: %d\n", a);
    printf("target 2: %d\n", b);
    printf("target 1: %d\n", c);
}

void func3() {

}

void func4(int a) {

}
