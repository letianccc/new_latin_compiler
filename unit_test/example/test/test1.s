//by latin

    .text
    .globl	_main
_main:
    pushl	%ebp
    pushl	%ebx
    pushl	%esi
    pushl	%edi
    movl	%esp, %ebp
    andl	$-16, %esp
    subl	$72, %esp
    movl	$3, %eax
    movl	%eax, 68(%esp)
    movl	$2, %eax
    movl	%eax, 64(%esp)
    movl	$1, %eax
    movl	%eax, 60(%esp)
    movl	$1, %eax
    movl	%eax, 56(%esp)
    movl	$3, %eax
    movl	%eax, 52(%esp)
    movl	60(%esp), %eax
    movl	%eax, 4(%esp)
    movl	$LC0, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$2, %eax
    movl	%eax, 48(%esp)
    movl	48(%esp), %eax
    movl	%eax, 4(%esp)
    movl	$LC1, %eax
    movl	%eax, 0(%esp)
    call	_printf
    fldl	FLOAT0
    fstpl	28(%esp)
    fldl	28(%esp)
    fstpl	4(%esp)
    movl	$LC2, %eax
    movl	%eax, 0(%esp)
    call	_printf
    fldl	FLOAT1
    fstpl	4(%esp)
    movl	$LC3, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$5, %eax
    movl	%eax, 4(%esp)
    movl	$LC4, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$LC5, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$2, %eax
    movl	%eax, 4(%esp)
    movl	$1, %eax
    movl	%eax, 0(%esp)
    call	_func1
    movl	$2, %eax
    movl	%eax, 4(%esp)
    movl	$1, %eax
    movl	%eax, 0(%esp)
    call	_func2
    call	_func3
    movl	$1, %eax
    movl	%eax, 0(%esp)
    call	_func4
    call	_getchar
    movl	%ebp, %esp
    popl	%edi
    popl	%esi
    popl	%ebx
    popl	%ebp
    ret
    .text
    .globl	_func1
_func1:
    pushl	%ebp
    pushl	%ebx
    pushl	%esi
    pushl	%edi
    movl	%esp, %ebp
    andl	$-16, %esp
    subl	$72, %esp
    movl	$3, %eax
    movl	%eax, 68(%esp)
    movl	$2, %eax
    movl	%eax, 64(%esp)
    movl	$1, %eax
    movl	%eax, 60(%esp)
    movl	$1, %eax
    movl	%eax, 56(%esp)
    movl	$3, %eax
    movl	%eax, 52(%esp)
    movl	60(%esp), %eax
    movl	%eax, 4(%esp)
    movl	$LC0, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$2, %eax
    movl	%eax, 48(%esp)
    movl	48(%esp), %eax
    movl	%eax, 4(%esp)
    movl	$LC1, %eax
    movl	%eax, 0(%esp)
    call	_printf
    fldl	FLOAT0
    fstpl	28(%esp)
    fldl	28(%esp)
    fstpl	4(%esp)
    movl	$LC2, %eax
    movl	%eax, 0(%esp)
    call	_printf
    fldl	FLOAT1
    fstpl	4(%esp)
    movl	$LC3, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$5, %eax
    movl	%eax, 4(%esp)
    movl	$LC4, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	$LC5, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	%ebp, %esp
    popl	%edi
    popl	%esi
    popl	%ebx
    popl	%ebp
    ret
    .text
    .globl	_func2
_func2:
    pushl	%ebp
    pushl	%ebx
    pushl	%esi
    pushl	%edi
    movl	%esp, %ebp
    andl	$-16, %esp
    subl	$12, %esp
    movl	20(%ebp), %eax
    movl	%eax, 8(%esp)
    movl	20(%ebp), %eax
    movl	%eax, 4(%esp)
    movl	$LC0, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	24(%ebp), %eax
    movl	%eax, 4(%esp)
    movl	$LC1, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	8(%esp), %eax
    movl	%eax, 4(%esp)
    movl	$LC0, %eax
    movl	%eax, 0(%esp)
    call	_printf
    movl	%ebp, %esp
    popl	%edi
    popl	%esi
    popl	%ebx
    popl	%ebp
    ret
    .text
    .globl	_func3
_func3:
    pushl	%ebp
    pushl	%ebx
    pushl	%esi
    pushl	%edi
    movl	%esp, %ebp
    andl	$-16, %esp
    subl	$0, %esp
    movl	%ebp, %esp
    popl	%edi
    popl	%esi
    popl	%ebx
    popl	%ebp
    ret
    .text
    .globl	_func4
_func4:
    pushl	%ebp
    pushl	%ebx
    pushl	%esi
    pushl	%edi
    movl	%esp, %ebp
    andl	$-16, %esp
    subl	$0, %esp
    movl	%ebp, %esp
    popl	%edi
    popl	%esi
    popl	%ebx
    popl	%ebp
    ret
LC0:
    .string	"target 1: %d\n"
LC1:
    .string	"target 2: %d\n"
LC2:
    .string	"target 3.5 : %f\n"
LC3:
    .string	"target 4.5 : %f\n"
LC4:
    .string	"target 5   : %d\n"
LC5:
    .string	"target None: None\n"
FLOAT0:
    .long	0
    .long	1074528256
FLOAT1:
    .long	0
    .long	1074921472
