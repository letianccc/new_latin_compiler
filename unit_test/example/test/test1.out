target 1: 1
target 2: 2
target 3.5 : 3.500000
target 4.5 : 4.500000
target 5   : 5
target None: None
target 1: 1
target 2: 2
target 3.5 : 3.500000
target 4.5 : 4.500000
target 5   : 5
target None: None
target 1: 1
target 2: 2
target 1: 1
